str.go

```diff
@@ -0,0 +1,85 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+)
+
+var (
+	ErrorBadEpoch = errors.New("[merkletree] Bad STR Epoch. STR's epoch must be nonzero")
+)
+
+// SignedTreeRoot represents a signed tree root, which is generated
+// at the beginning of every epoch.
+// Signed tree roots contain the current root node,
+// the current and previous epochs, the hash of the
+// previous STR, and its signature.
+// STR should be final
+type SignedTreeRoot struct {
+	tree        *MerkleTree
+	epoch       uint64
+	prevEpoch   uint64
+	prevStrHash []byte
+	sig         []byte
+	policies    Policies
+}
+
+func NewSTR(key crypto.SigningKey, policies Policies, m *MerkleTree, epoch uint64, prevHash []byte) *SignedTreeRoot {
+	if epoch < 1 {
+		panic(ErrorBadEpoch)
+	}
+	str := &SignedTreeRoot{
+		tree:        m,
+		epoch:       epoch,
+		prevEpoch:   epoch - 1,
```

Comment:
I'm actually not sure if this is what the paper is implying (could have just omitted it to make the figure look a bit cleaner). But because of https://github.com/coniks-sys/libmerkleprefixtree-go/pull/8#discussion_r70645139, I think it's fine to omit this value.